### ISSUE TITLE
Octokit Auth

### ISSUE THREAD

**Issue by Peter554** [Peter554/github-repo-tagger#1]
Can someone who knows more about it please comment on the octokit auth (https://github.com/octokit/rest.js/) and whether it is being used correctly/safely? Would it be safer to use the OAuth option and if so is this any easy change?

**Comment by Peter554** [comment]
As I release it as a 'static site' via Github pages and the code is open source - you can see yourself I do nothing malicious with user credentials -  I think what I've done should be safe? I followed the guidance given at the link above (here again - https://github.com/octokit/rest.js/).

Potentially someone who knows more could look into building this into an official GitHub App (https://developer.github.com/apps/)?

**Comment by Thatkookooguy** [comment]
@Peter554 Hi Peter! I'll try and take a look at the auth you implemented.

I wanted to open a new issue that this doesn't support **two factor authentication**, but I guess it's related to this so I'll take a look :-)

**Comment by Peter554** [comment]
Thanks @Thatkookooguy, yes I also realised this recently as I've had to add 2 factor authentication to my account and now this doesn't work any more for me!!

**Comment by Peter554** [comment]
@Thatkookooguy I made some substantial changes in [this commit](https://github.com/Peter554/github-repo-tagger/commit/8c4c12efba485df7836c2accb1550b54a969038a) that relate to this issue. I've now moved to using the github apps oauth workflow and have a simple backend hosted on heroku to handle this. I'm still not 100% what I'm doing is correct so if you or anyone else wants to review that would still be much appreciated.
